src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/Utf8StringMarshaller.cs

```diff
@@ -101,5 +103,26 @@ public void FreeNative()
             if (_allocated != null)
                 Marshal.FreeCoTaskMem((IntPtr)_allocated);
         }
+
+        internal static string? ConvertToManaged(byte* value)
+            => Marshal.PtrToStringUTF8((IntPtr)value);
+
+        internal static byte* ConvertToNative(string? str)
+        {
+            // This is functionally equivalent to Marshal.StringToCoTaskMemUTF8 except the conservative over-allocations
```

Comment:
This again raising the question about over-allocation. Notice that the existing built-in Utf8 string marshaller is not over-allocating for this case.